services/repository/src/test/java/org/sagebionetworks/repo/web/filter/UserThrottleFilterTest.java

```diff
@@ -1,102 +1,161 @@
+		try {
+			filter.doFilter(request, response, filterChain);
+		} finally {
+			verify(userThrottleGate).attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS);
+			verifyNoMoreInteractions(filterChain, userThrottleGate, userFrequencyThrottleGate);
 		}
 	}
 
 	@Test
-	public void testNoEmptySlots() throws Exception {
-		MockHttpServletRequest request = new MockHttpServletRequest();
-		request.setParameter(AuthorizationConstants.USER_ID_PARAM, "111");
-		MockHttpServletResponse response = new MockHttpServletResponse();
-		FilterChain filterChain = mock(FilterChain.class);
+	public void testNoEmptyConcurrentConnectionSlots() throws Exception {
 		Consumer consumer = mock(Consumer.class);
 		ReflectionTestUtils.setField(filter, "consumer", consumer);
 
-		when(userThrottleGate.attemptToAcquireLock("111", LOCK_TIMOUTE_SEC, MAX_CONCURRENT_LOCKS)).thenReturn(null);
+		when(userThrottleGate.attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS)).thenReturn(null);
+		when(userFrequencyThrottleGate.attemptToAcquireLock(userId, REQUEST_FREQUENCY_LOCK_TIMEOUT_SEC, MAX_REQUEST_FREQUENCY_LOCKS)).thenReturn(true);
 
+		
 		filter.doFilter(request, response, filterChain);
-		assertEquals(503, response.getStatus());
+		assertEquals(HttpStatus.SC_SERVICE_UNAVAILABLE, response.getStatus());
+		
+		ArgumentCaptor<ProfileData> profileDataArgument = ArgumentCaptor.forClass(ProfileData.class); 
+		verify(consumer).addProfileData(profileDataArgument.capture());
+		assertEquals("ConcurrentConnectionsLockUnavailable", profileDataArgument.getValue().getName());
+		
+
+		verify(userThrottleGate).attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS);
+		verify(consumer).addProfileData(any(ProfileData.class));
+		verifyNoMoreInteractions(filterChain, userThrottleGate, userFrequencyThrottleGate, consumer);
+	}
+	
+	@Test
+	public void testNoEmptyRequestFrequencySlots() throws Exception {
+		Consumer consumer = mock(Consumer.class);
+		ReflectionTestUtils.setField(filter, "consumer", consumer);
+		
+		when(userThrottleGate.attemptToAcquireLock(userId, CONCURRENT_CONNECTIONS_LOCK_TIMEOUT_SEC, MAX_CONCURRENT_LOCKS)).thenReturn(concurrentSemaphoreToken);
+		when(userFrequencyThrottleGate.attemptToAcquireLock(userId, REQUEST_FREQUENCY_LOCK_TIMEOUT_SEC, MAX_REQUEST_FREQUENCY_LOCKS)).thenReturn(false);
 
-		verify(userThrottleGate).attemptToAcquireLock("111", LOCK_TIMOUTE_SEC, MAX_CONCURRENT_LOCKS);
+		filter.doFilter(request, response, filterChain);
+		assertEquals(HttpStatus.SC_SERVICE_UNAVAILABLE, response.getStatus());
+		
+		ArgumentCaptor<ProfileData> profileDataArgument = ArgumentCaptor.forClass(ProfileData.class); 
+		verify(consumer).addProfileData(profileDataArgument.capture());
+		assertEquals("RequestFrequencyLockUnavailable", profileDataArgument.getValue().getName());
```

Comment:
make RequestFrequencyLockUnavailable a constant in UserThrottleFilter and refer to it here